[Enhancement] Change the parameter strict_mode to null_if_not_found of dict_mapping function(#40897)

srlch opened the pull request:
The last optional parameter for dict_mapping is strict_mode(boolean), and its behavior if following:

If strict_mode is True : return an exception if the key is not found in dictionary table.
If strict_mode is False(Default) : return null if the key is not found in dictionary table.

We change the strict_mode into null_if_not_found to make more clear for user to use this function, and the behavior of null_if_not_found is following:
If null_if_not_found is True : return null if the key is not found in dictionary table.
If null_if_not_found is False(Default) : return an exception if the key is not found in dictionary table.

Fixes #40897

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [x] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

trueeyu commented on be/src/exprs/dict_query_expr.cpp:
```diff
@@ -33,6 +33,7 @@ DictQueryExpr::DictQueryExpr(const TExprNode& node) : Expr(node), _dict_query_ex
 StatusOr<ColumnPtr> DictQueryExpr::evaluate_checked(ExprContext* context, Chunk* ptr) {
     Columns columns(children().size());
     size_t size = ptr != nullptr ? ptr->num_rows() : 1;
+    bool null_if_not_found = !_dict_query_expr.strict_mode;
```
If later insert_to_dict_table_if_not_found, how to name the session variables.

trueeyu replied:
null_if_not_found, insert_if_not_found, exception_if_not_found ?

srlch committed:
[Enhancement] Change the parameter strict_mode to null_if_not_found of dict_mapping function(#40897)

The last optional parameter for dict_mapping is strict_mode(boolean), and its behavior
if following:
If strict_mode is True : return an exception if the key is not found in dictionary table.
If strict_mode is False(Default) : return null if the key is not found in dictionary table.

We change the strict_mode into null_if_not_found to make more clear for user to use this function,
and the behavior of null_if_not_found is following:
If null_if_not_found is True : return null if the key is not found in dictionary table.
If null_if_not_found is False(Default) : return an exception if the key is not found in dictionary table.

Signed-off-by: srlch <linzichao@starrocks.com>
